wadsrc/static/zscript/doom/bossbrain.txt

```diff
@@ -165,23 +165,28 @@ extend class Actor
 			boom.tics -= random[BrainScream](0, 7);
 			if (boom.tics < 1) boom.tics = 1;
 		}
+		else boom.Destroy(); //destroy the actor if it doesn't have BrainExplode state.
 	}
 	
-	void A_BrainScream()
+	void A_BrainScream(class <Actor> explosionactortype = "Rocket")
 	{
+		//[Cacodemon345] validate the given string.
+		if (explosionactortype == null) explosionactortype = "Rocket";
 		for (double x = -196; x < +320; x += 8)
 		{
 			// (1 / 512.) is actually what the original value of 128 did, even though it probably meant 128 map units.
-			BrainishExplosion(Vec2OffsetZ(x, -320, (1 / 512.) + random[BrainExplode](0, 255) * 2));
+			BrainishExplosion(Vec2OffsetZ(x, -320, (1 / 512.) + random[BrainExplode](0, 255) * 2), explosionactortype =  "Rocket");
 		}
 		A_PlaySound("brain/death", CHAN_VOICE, 1, false, ATTN_NONE);
 	}
 	
-	void A_BrainExplode()
+	void A_BrainExplode(class <Actor> explosionactortype = "Rocket")
 	{
+		//[Cacodemon345] validate the given string.
```

Comment:
... and this one too.
These comments describe obvious things.

Moreover, these checks are completely useless.
Does it abort without them? No.
When `null` is passed instead of class and the default value isn't `null` it's fine if these functions do nothing.